Classes/FormElements/Recaptcha.php

```diff
@@ -71,44 +73,47 @@ public function onSubmit(FormRuntime $formRuntime, &$elementValue)
             return;
         }
 
-        $requestMethodString = strtolower($this->settings['requestMethod']);
-        if ($requestMethodString === 'curl') {
-            $requestMethod = new \ReCaptcha\RequestMethod\CurlPost();
-        } elseif ($requestMethodString === 'socket') {
-            $requestMethod = new \ReCaptcha\RequestMethod\SocketPost();
+        $requestMethodString = $this->settings['requestMethod'];
+
+        if (self::isReCaptchaRequestClass($requestMethodString)) {
+            $requestMethod = new $requestMethodString();
         } else {
-            $requestMethod = new \ReCaptcha\RequestMethod\Post();
+            $requestMethod = match (strtolower($requestMethodString)) {
```

Comment:
When using `match`, please add `"php": ">= 8.0"` to the composer dependencies. Otherwise we could move this part into a function:

```php
public function onSubmit(FormRuntime $formRuntime, &$elementValue)
{
    // Previous code
    $requestMethod = $this->getRequestMethod($requestMethodString);
    // Code after
}

/**
 * Get request method from string
 * @param string $requestMethod
 * @return \ReCaptcha\RequestMethod
 */
protected function getRequestMethod(string $requestMethodString): \ReCaptcha\RequestMethod
{
    if (self::isReCaptchaRequestClass($requestMethodString)) {
        return new $requestMethodString();
    }
    
    $requestMethodString = strtolower($requestMethodString);
    if ($requestMethodString === 'curl') {
        return new \ReCaptcha\RequestMethod\CurlPost();
    }
    
    if ($requestMethodString === 'socket') {
        return new \ReCaptcha\RequestMethod\SocketPost();
    }
    
    return new \ReCaptcha\RequestMethod\Post();
}
```